### PR TITLE
MAINT Use list and dict comprehension

### PR DESCRIPTION
**Pull request by rth**
This uses list and dict comprehension in places where it doesn't hurt readability. It should be a bit faster and less verbose.

### CODE REVIEW

File: pandas/io/formats/printing.py [hunk 1]
```diff
@@ -110,10 +110,11 @@ def _pprint_seq(seq, _nest_lvl=0, max_seq_items=None, **kwds):
...
 
     s = iter(seq)
-    r = []
-    for i in range(min(nitems, len(seq))):  # handle sets, no slicing
-        r.append(pprint_thing(
-            next(s), _nest_lvl + 1, max_seq_items=max_seq_items, **kwds))
+    # handle sets, no slicing
+    r = [
+        pprint_thing(
+            next(s), _nest_lvl + 1, max_seq_items=max_seq_items, **kwds)
```
Comment (datapythonista) [inline pandas/io/formats/printing.py]: personally I think it'd be a bit more readable like:
```
r = [pprint_thing(next(s),
                  _nest_lvl + 1,
...
```

**Review by datapythonista — APPROVED** [review]
Nice fixes, thanks for taking care of them.

Just added a comment that IMO would improve readability, but lgtm.

**Comment by jreback** [comment]
can you merge master, ping on green.

**Comment by rth** [comment]
Thanks @datapythonista ! Addressed your comment.

@jreback Merged master in and CI is green.

**Comment by jreback** [comment]
thanks @rth

**Pull request merged**